yaml/roads.yaml

```diff
@@ -202,25 +208,42 @@ filters:
       highway: {col: highway}
   - filter:
       highway: [pedestrian, track, cycleway]
-    min_zoom: 13
+    min_zoom: 'LEAST(mz_calculate_path_major_route(osm_id), 13)'
     table: osm
     output:
       <<: *osm_highway_properties
       <<: *osm_network_from_tags
       kind: path
       highway: {col: highway}
   - filter:
-      highway: [path, footway, steps]
-    min_zoom: |
-      COALESCE(mz_calculate_path_major_route(osm_id),
-        CASE
-          WHEN highway = 'path' THEN 13
-          WHEN mz_is_path_named_or_designated(highway, name, bicycle, foot,
```

Comment:
I made some further changes from this branch in https://github.com/mapzen/vector-datasource/commits/nvkelso/early-path-revisions a couple days ago. I'll take on merging those concepts & updated tests into this PR. Thanks!